source/contracts/reporting/Auction.sol

```diff
@@ -0,0 +1,289 @@
+        initializeNewAuctionIfNeeded();
+        require(!bootstrapMode);
+        uint256 _timePassed = controller.getTimestamp().sub(initializationTime).sub(currentAuctionIndex * 1 days);
+        uint256 _priceDecrease = initialEthSalePrice.mul(_timePassed) / Reporting.getAuctionDuration();
+        return initialEthSalePrice.sub(_priceDecrease);
+    }
+
+    function getCurrentAttoRepBalance() public returns (uint256) {
+        uint256 _repSalePriceInAttoEth = getRepSalePriceInAttoEth();
+        uint256 _ethSupply = repAuctionToken.totalSupply();
+        uint256 _attoRepSold = _ethSupply.mul(10**18).div(_repSalePriceInAttoEth);
+        if (_attoRepSold >= initialAttoRepBalance) {
+            return 0;
+        }
+        return initialAttoRepBalance.sub(_attoRepSold);
+    }
+
+    function getCurrentAttoEthBalance() public returns (uint256) {
+        uint256 _ethSalePriceInAttoRep = getEthSalePriceInAttoRep();
+        uint256 _repSupply = ethAuctionToken.totalSupply();
+        uint256 _attoEthSold = _repSupply.mul(10**18).div(_ethSalePriceInAttoRep);
+        if (_attoEthSold >= initialAttoEthBalance) {
+            return 0;
+        }
+        return initialAttoEthBalance.sub(_attoEthSold);
+    }
+
+    function getDerivedRepPriceInAttoEth() public view returns (uint256) {
+        if (repAuctionToken == IAuctionToken(0) || ethAuctionToken == IAuctionToken(0)) {
+            return repPrice;
+        }
+        uint256 _repAuctionTokenMaxSupply = repAuctionToken.maxSupply();
+        uint256 _ethAuctionTokenMaxSupply = ethAuctionToken.maxSupply();
+        if (_repAuctionTokenMaxSupply == 0 || _ethAuctionTokenMaxSupply == 0) {
+            return repPrice;
+        }
+        uint256 _upperBoundRepPrice = repAuctionToken.maxSupply().mul(10**18).div(initialAttoRepBalance);
+        uint256 _lowerBoundRepPrice = initialAttoEthBalance.mul(10**18).div(_ethAuctionTokenMaxSupply);
+        return _upperBoundRepPrice.add(_lowerBoundRepPrice) / 2;
+    }
+
+    function getRepPriceInAttoEth() public view returns (uint256) {
+        // Use the manually provided REP price if the Auction system is not used for setting fees yet
+        if (!controller.useAuction()) {
+            return manualRepPriceInAttoEth;
+        }
+
+        // If this auction is over and it is a recorded auction use the price it found
+        if (getAuctionIndexForCurrentTime() != currentAuctionIndex && currentRoundType == RoundType.RECORDED) {
+            return getDerivedRepPriceInAttoEth();
+        }
+
+        return repPrice;
+    }
+
+    function getAuctionIndexForCurrentTime() public view returns (uint256) {
+        return controller.getTimestamp().sub(initializationTime) / Reporting.getAuctionDuration();
+    }
+
+    function isActive() public view returns (bool) {
+        RoundType _roundType = getRoundType();
+        return _roundType == RoundType.UNRECORDED || _roundType == RoundType.RECORDED;
```

Comment:
What if it's in bootstrapMode?

Reply:
Hm so for the case where people don't participate and it ends up occuring on the RECORDED round perhaps we could ignore it explicitly.

Thing is we wont be using the results anyway for some time so I'm not sure its worth the code.